internal/backend/db/dbgorm/ownership.go

```diff
@@ -120,40 +121,52 @@ func (gdb *gormdb) createEntityWithOwnership(
 	}
 
 	return gdb.transaction(ctx, func(tx *tx) error {
-		if err := tx.isUserEntity(user, idsToVerifyOwnership...); err != nil {
+		if err := tx.isUserEntity(uid, idsToVerifyOwnership...); err != nil {
 			return err
 		}
-		if err := create(tx.db, user); err != nil { // create
+		if err := create(tx.db, uid); err != nil { // create
 			return err
 		}
-		return saveOwnershipForEntities(tx.db, user, ownerships...) // ensure user and add ownerships
+		return saveOwnershipForEntities(tx.db, uid, ownerships...) // ensure user and add ownerships
 	})
 }
 
-func isUserEntity(db *gorm.DB, uid string, ids ...sdktypes.UUID) error {
+func getOwnerships(db *gorm.DB, ids ...sdktypes.UUID) ([]scheme.Ownership, error) {
 	if len(ids) == 0 {
-		return nil
+		return nil, nil
 	}
 
-	var oo []scheme.Ownership
-	if err := db.Model(&scheme.Ownership{}).Where("entity_id IN ?", ids).Select("user_id").Find(&oo).Error; err != nil {
-		return err
-	}
-
-	if len(oo) < len(ids) {
-		return gorm.ErrRecordNotFound
+	var ownerships []scheme.Ownership
+	if err := db.Model(&scheme.Ownership{}).Where("entity_id IN ?", ids).Find(&ownerships).Error; err != nil {
+		return nil, err
 	}
+	return ownerships, nil
+}
 
-	for _, o := range oo {
+func verifyOwnerships(uid string, ownerships []scheme.Ownership) error {
+	for _, o := range ownerships {
 		if o.UserID != uid {
 			return sdkerrors.ErrUnauthorized
 		}
 	}
 	return nil
 }
 
+// fetches ownerships for given ids and verifies that user have access to all of them
+func ensureUserAccessToEntitiesWithOwnerships(db *gorm.DB, uid string, ids ...sdktypes.UUID) ([]scheme.Ownership, error) {
+	ownerships, err := getOwnerships(db, ids...)
+	if err != nil {
+		return ownerships, err
+	}
+
+	if len(ownerships) < len(ids) { // should be equal
+		return ownerships, gorm.ErrRecordNotFound
+	}
+	return ownerships, verifyOwnerships(uid, ownerships)
```

Comment:
this is error prone if you return an error and someone forget to check it, 
they would still have a list of ownerships which looks like a valid thing, 
probably should return null if there is an error

Reply:
Disagree actually. according to the go conventions you should check the error. 
The reason for this that error could be notFOund, e.g. we fetch less ownerships then requested. But maybe it's ok for any specific case, so I would like to return an error and return data fetched.

Reply:
rewritten. 
Now it returns nil in all cases when I cannot guarantee that results are valid.
Only if case when there are less ownerships then requested found it will return NotFound error. But in this case it's guaranteed that returned results are valid. So it's caller decision how and when use read results in case of NotFound error. Note that this is similar to what Reader.Read is doing - when some bytes are read, but less than requested. So you could know both that there was some problem and you could know how many bytes read.

Reply:
@tebeka , could u plz advise on this one?
is it OK to return an error along valid, but less expected result?

Reply:
acking this one for now, 
@tebeka will have a look later and in case majority thinks current solution is fine, we're done, otherwise we would need to fix it in a different PR
not resolving this issue but approving the PR

Reply:
fixed